colcon_core/package_descriptor.py

```diff
@@ -100,24 +101,24 @@ def get_recursive_dependencies(
         dependencies = set()
         while queue:
             # pick one dependency from the queue
-            name = queue.pop()
+            dep = queue.pop()
             # ignore redundant dependencies
-            if name in dependencies:
+            if dep in dependencies:
                 continue
             # ignore circular dependencies
-            if name == self.name:
+            if dep == self.name:
```

Comment:
Should this be changed to `dep.name` then - now that we ensure each `dep` is actually a `DependencyDescriptor`?

Reply:
So I did this, and the performance got really bad - about a 100% increase in listing times. Probably caused by the string reallocation in `DependencyDescriptor.name`. I had an idea last night on how we could mitigate that - let me do some tests.

Reply:
Looks like it wasn't as bad as I thought, but still not great. Just making this changes yields a 65% increase in listing times.

The following change to `dependency_descriptor.py` brings the loss down to a 23% increase, but duplicates the `name` value:
```patch
@@ -22,15 +22,4 @@ class DependencyDescriptor(str):
 
     def __init__(self, name, *, metadata=None):  # noqa: D107
         self.metadata = metadata if metadata is not None else {}
-
-    @property
-    def name(self):
-        """
-        Name of the dependency.
-
-        The property exists for future compatibility when the base class is
-        removed.
-
-        :rtype: str
-        """
-        return str(self)
+        self.name = name
```

I'm out of ideas to mitigate the losses. Any input would be appreciated.

Reply:
We can also leave the comparison as is - it was just an idea to make it clearer what was actually compared.